kafka/consumer.c

```diff
@@ -43,9 +44,28 @@ consumer_poll_loop(void *arg) {
             rd_msg = rd_kafka_consumer_poll(poller->rd_consumer, 1000);
             if (rd_msg != NULL) {
                 if (rd_msg->err != RD_KAFKA_RESP_ERR_NO_ERROR) {
-                    // FIXME: push errors to error queue?
+                    errors_count++;
+
+                    error_msg_t *msg = new_error_msg(rd_msg->err, rd_kafka_err2str(rd_msg->err));
+
+                    // free rdkafka message instantly to prevent hang on close / destroy consumer
                     rd_kafka_message_destroy(rd_msg);
+                    rd_msg = NULL;
+
+                    pthread_mutex_lock(&event_queues->queues[ERROR_QUEUE]->lock);
+
+                    queue_lockfree_push(event_queues->queues[ERROR_QUEUE], msg);
+                    count = event_queues->queues[ERROR_QUEUE]->count;
+
+                    pthread_mutex_unlock(&event_queues->queues[ERROR_QUEUE]->lock);
```

Comment:
I guess you don't need to acquire lock if you will use "queue_push" instead of "queue_lockfree_push"

Reply:
Since the `count` variable is not used, then yes